Add license to project

lonvia opened the pull request:
As discussed in #60, this adds a license to the project. The basic license is MIT. In addition, there is a special exception in the README that puts all tests into the public domain to make them as reusable as possible.

I would kindly request that all that have contributed so far explicitly approve this PR if the license is fine with you. In particular, please make sure that tests that you have contributed can be put into the public domain. If you have reservations or questions, please comment on #60 or get in touch with me directly.

Contributors list according to github: @yohanboniface @nlehuby @hbruch @karussell @kinnou02 @antoine-de @jfgigand 
Please approve this PR or comment, if this change is okay with you.

nlehuby commented on README.md:
```diff
@@ -118,3 +118,12 @@ has the subkeys you want to test against (`name`, `housenumber`…).
+
+As a special exception, the test cases under `geocoder_tester/world/` are
+considered to be in the public domain. You may use them without any
+restrictions.
```
I've used 3rd parties datasets (Wikidata, OpenFlights, etc) to create some test cases, so some of them may not be considered to be in the public domain. For instance: https://github.com/geocoders/geocoder-tester/blob/master/geocoder_tester/world/poi/license.md
Maybe add "unless otherwise specified in the test directory readme" here ?

lonvia commented:
I have now changed the text as per @nlehuby  suggest and added an explicit attribution for wikidata and OpenFlights.

However, this makes me realise that quite a bit of the test data is based on OpenStreetMap data, probably more than what would allow us to claim it as an insubstantial abstract.
